File: app/src/main/java/com/baronzhang/android/weather/view/selectcity/SelectCityContract.java
package com.baronzhang.android.weather.view.selectcity;

import java.util.List;

import com.baronzhang.android.weather.model.db.entities.City;
import com.baronzhang.android.weather.view.main.BasePresenter;
import com.baronzhang.android.weather.view.main.BaseView;

/**
 * @author baronzhang (baron[dot]zhanglei[at]gmail[dot]com)
 */
public interface SelectCityContract {

    interface View extends BaseView<Presenter> {

        void displayCities(List<City> cities);
    }

    interface Presenter extends BasePresenter {

        void loadCities();
    }
}

File: weather-mvp-dagger-kotlin/src/main/java/com/minimalist/weather/kotlin/view/contract/SelectCityContract.kt
package com.minimalist.weather.kotlin.view.contract


import com.minimalist.weather.kotlin.model.data.entity.city.City

interface SelectCityContract {

    interface View : BaseView<Presenter> {

        fun displayCities(cities: List<City>)
    }

    interface Presenter : BasePresenter {

        fun loadCities()
    }
}
